packages/material-ui-unstyled/src/SwitchUnstyled/SwitchUnstyled.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { ComponentPropsWithoutRef, CSSProperties, ElementType } from 'react';
+import clsx from 'clsx';
+import useSwitch, { UseSwitchProps } from './useSwitch';
+import classes from './switchUnstyledClasses';
+import ButtonUnstyled from '../ButtonUnstyled/ButtonUnstyled';
+
+export interface SwitchUnstyledProps<
+  TRoot extends ElementType,
+  TThumb extends ElementType,
+  TInput extends ElementType,
+> extends UseSwitchProps {
+  components?: {
+    Root?: TRoot;
+    Thumb?: TThumb;
+    Input?: TInput;
+  };
+  componentsProps?: {
+    root?: ComponentPropsWithoutRef<TRoot>;
+    thumb?: ComponentPropsWithoutRef<TThumb>;
+    input?: ComponentPropsWithoutRef<TInput>;
+  };
+}
+
+const minimalRootStyles: CSSProperties = {
```

Comment:
```suggestion
const baseRootStyles: CSSProperties = {
```
Or maybe even:
```suggestion
const rootStyles: CSSProperties = {
```

As we are already in an `Unstyled*` component.